src/g1.rs

```diff
@@ -312,7 +312,14 @@ impl AffineRepr for G1Affine {
     }
 
     fn from_random_bytes(bytes: &[u8]) -> Option<Self> {
-        Option::from(blstrs::G1Affine::from_uncompressed(bytes.try_into().unwrap()).map(Self))
+        let mut b = [0u8; 32];
+        b.copy_from_slice(bytes);
+        Option::from(
+            G1Projective(<blstrs::G1Projective as group::Group>::random(
+                rand::rngs::StdRng::from_seed(b),
+            ))
+            .into_affine(),
+        )
```

Comment:
This is using the bytes just as random seed. I think the purpose of this function is, that if you have a (valid) set of bytes, it will create an element out of those bytes. It can e.g. be used to get a field element from zcash based crates to the arkworks one.

Reply:
It's not meant to sample a random point. That's what `rand()` is for.